src/libostree/ostree-repo-checkout.c

```diff
@@ -643,6 +647,32 @@ checkout_one_file_at (OstreeRepo                        *repo,
 
       need_copy = FALSE;
     }
+  else if (is_overlayfs_whiteout)
+    {
+      const char *name = destination_name + (sizeof (OVERLAYFS_WHITEOUT_PREFIX) - 1);
+
+      if (!name[0])
+        return glnx_throw (error, "Invalid empty overlayfs whiteout '%s'", name);
+
+      g_assert (name[0] != '/'); /* Sanity */
+
+      if (mknodat(destination_dfd, name, (source_mode & ~S_IFMT) | S_IFCHR, (dev_t)0) < 0)
+          return glnx_throw_errno_prefix (error, "Creating whiteout char device");
+      if (options->mode != OSTREE_REPO_CHECKOUT_MODE_USER)
+        {
+          if (source_xattrs != NULL &&
+              !glnx_dfd_name_set_all_xattrs(destination_dfd, name, source_xattrs,
+                                              cancellable, error))
+              return glnx_throw_errno_prefix (error, "Setting xattrs for whiteout char device");
+
+          if (TEMP_FAILURE_RETRY(fchownat(destination_dfd, name,
+                                          g_file_info_get_attribute_uint32 (source_info, "unix::uid"),
+                                          g_file_info_get_attribute_uint32 (source_info, "unix::gid"),
+                                          AT_SYMLINK_NOFOLLOW) < 0))
+              return glnx_throw_errno_prefix (error, "fchownat");
+        }
+      need_copy = FALSE;
```

Comment:
We could probably just `return TRUE;` here too - what we're skipping over is the caching stuff which isn't relevant for this.